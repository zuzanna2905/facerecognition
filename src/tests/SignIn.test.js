import {shallow} from 'enzyme';
import React from 'react';
import SignIn from '../component/SignIn/SignIn';

it('renders SignIn without crashing', () => {
    let wrapper = shallow(<SignIn />);
    expect(wrapper).toMatchSnapshot();
});

it('simulate SignIn', () => {    
    const expectedState = {
        signInEmail: 'john@gmail.com',
        signInPassword: 'cookies'
    }
    let wrapper2 = shallow(<SignIn />);
    wrapper2.find('[id="password"]').simulate('change', {target: {value: 'cookies'}});
    wrapper2.find('[id="email-address"]').simulate('change', {target: {value: 'john@gmail.com'}});
    expect(wrapper2.state().password).toEqual(expectedState.password);
    expect(wrapper2.state().email).toEqual(expectedState.email);
    wrapper2.find('[id="signinsubmit"]').simulate('click');  
})