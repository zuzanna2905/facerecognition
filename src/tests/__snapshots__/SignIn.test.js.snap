// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders SignIn without crashing 1`] = `
<article
  className="br3 ba dark-gray b--black-10 mv4 w-100 w-50-m w-25-l mw6 shadow-5 center"
>
  <main
    className="pa4 black-80"
  >
    <div
      className="measure"
    >
      <fieldset
        className="ba b--transparent ph0 mh0"
        id="sign_up"
      >
        <legend
          className="f2 fw6 ph0 mh0"
        >
          Sign In
        </legend>
        <div
          className="mt3"
        >
          <label
            className="db fw6 lh-copy f6"
            htmlFor="email-address"
          >
            Email
          </label>
          <input
            className="pa2 input-reset ba bg-transparent hover-bg-black hover-white w-100"
            id="email-address"
            name="email-address"
            onChange={[Function]}
            type="email"
          />
        </div>
        <div
          className="mv3"
        >
          <label
            className="db fw6 lh-copy f6"
            htmlFor="password"
          >
            Password
          </label>
          <input
            className="b pa2 input-reset ba bg-transparent hover-bg-black hover-white w-100"
            id="password"
            name="password"
            onChange={[Function]}
            type="password"
          />
        </div>
      </fieldset>
      <div
        className=""
      >
        <input
          className="b ph3 pv2 input-reset ba b--black bg-transparent grow pointer f6 dib"
          id="signinsubmit"
          onClick={[Function]}
          type="submit"
          value="Sign in"
        />
      </div>
      <div
        className="lh-copy mt3"
      >
        <p
          className="f6 link dim black db"
          onClick={[Function]}
        >
          Register
        </p>
      </div>
    </div>
  </main>
</article>
`;
